platforms/android/demo/src/main/java/com/mapzen/tangram/android/MainActivity.java

```diff
@@ -31,36 +40,130 @@
 
 import java.io.File;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class MainActivity extends Activity implements OnMapReadyCallback, TapResponder,
-        DoubleTapResponder, LongPressResponder, FeaturePickListener, LabelPickListener, MarkerPickListener, SceneUpdateErrorListener {
+public class MainActivity extends AppCompatActivity implements OnMapReadyCallback, TapResponder,
+        DoubleTapResponder, LongPressResponder, FeaturePickListener, LabelPickListener, MarkerPickListener, SceneUpdateErrorListener,
+        AdapterView.OnItemSelectedListener, AdapterView.OnItemClickListener {
+
+    private final String apiKey = "vector-tiles-tyHL4AY";
+    private ArrayList<SceneUpdate> sceneUpdates = new ArrayList<>();
+
+    /* Override ArrayAdaptor to include any text */
+    private class CustomArrayAdapter extends ArrayAdapter {
+
+        List<String> suggestions;
+
+        Filter nameFilter = new Filter() {
+            @Override
+            public CharSequence convertResultToString(Object resultValue) {
+                return resultValue.toString().toLowerCase();
+            }
+
+            @Override
+            protected FilterResults performFiltering(CharSequence constraint) {
+                if (constraint != null) {
+                    suggestions.clear();
+                    for (String scene: scenes) {
+                        if (scene.toLowerCase().contains(constraint.toString().toLowerCase())) {
+                            suggestions.add(scene);
+                        }
+                    }
+                    if (suggestions.isEmpty()) {
+                        suggestions.add(constraint.toString().toLowerCase()); //include the text when nothing matches
+                    }
+
+                    FilterResults filterResults = new FilterResults();
+                    filterResults.values = suggestions;
+                    filterResults.count = suggestions.size();
+                    return filterResults;
+                } else {
+                    return new FilterResults();
+                }
+            }
+
+            @Override
+            protected void publishResults(CharSequence constraint, FilterResults results) {
+                List<String> filterList = (ArrayList<String>) results.values;
+                if (results != null && results.count > 0) {
+                    clear();
+                    for (String scene: filterList) {
+                        add(scene);
+                        notifyDataSetChanged();
+                    }
+                }
+            }
+        };
+
+        CustomArrayAdapter(Context context, int resource, List<String> objects) {
```

Comment:
I think you can do that. I get an `java.lang.UnsupportedOperationException`. Checked a couple of stackoverflows and they say because one can not `clear` the contents as is being done in the custom filter here. Of course if I am missing something java(y) then I am not sure.